portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -22,30 +22,45 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
-  private List<String> messages;
-
-  @Override
-  public void init() {
-    messages = new ArrayList<>();
-  }
   
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
+    Query query = new Query("Comment");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    
+    List<String> comments = new ArrayList<>();
+    for (Entity entity: results.asIterable()) {
+        String comment = (String) entity.getProperty("commentContent");
+        comments.add(comment);
+    }
+    
+    response.setContentType("application/json;");
     Gson gson = new Gson();
-    String json = gson.toJson(messages);
+    String json = gson.toJson(comments);
     response.getWriter().println(json);
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String comment = request.getParameter("comments-input");
-    messages.add(comment);
+    Entity commentEntity = new Entity("Comment");
```

Comment:
Since the entity and property names ("Comment" and "commentContent") are referenced in multiple places that need to stay in-sync, it's best to make them constants (ensures that typos are caught at compile time).

Reply:
Should I be declaring the entity and property names as constants at the class level and then use those constants in both the doPost and doGet methods?

Reply:
Yep, looks good!